scripts/triggers-end-to-end-tests/tests.inspect.ts

```diff
@@ -0,0 +1,72 @@
+import { expect } from "chai";
+import * as fs from "fs";
+import * as path from "path";
+
+import { FrameworkOptions, TriggerEndToEndTest } from "../integration-helpers/framework";
+
+const FIREBASE_PROJECT = process.env.FBTOOLS_TARGET_PROJECT || "";
+/*
+ * Various delays that are needed because this test spawns
+ * parallel emulator subprocesses.
+ */
+const TEST_SETUP_TIMEOUT = 80000;
+const EMULATORS_SHUTDOWN_DELAY_MS = 5000;
+
+function readConfig(): FrameworkOptions {
+  const filename = path.join(__dirname, "firebase.json");
+  const data = fs.readFileSync(filename, "utf8");
+  return JSON.parse(data);
+}
+
+describe("function triggers with inspect flag", () => {
+  let test: TriggerEndToEndTest;
+
+  before(async function (this) {
+    this.timeout(TEST_SETUP_TIMEOUT);
+
+    expect(FIREBASE_PROJECT).to.exist.and.not.be.empty;
+
+    const config = readConfig();
+    test = new TriggerEndToEndTest(FIREBASE_PROJECT, __dirname, config);
+    await test.startEmulators(["--only", "functions", "--inspect-functions"]);
+  });
+
+  after(async function (this) {
+    this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+    await test.stopEmulators();
+  });
+
+  it("should invoke correct function in the same codebase", async function (this) {
```

Comment:
Nit (feel free to ack):

is there a reason for using `function` instead of phat arrows?

Reply:
yes! Turns out, `this` isn't available if you use arrow fns.